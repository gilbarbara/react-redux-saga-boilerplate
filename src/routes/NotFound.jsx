import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { Container, Heading } from 'styled-minimal';
import Background from 'components/Background';

const StyledContainer = styled(Container)`
  align-items: center;
  text-align: center;

  h1,
  a {
    color: #fff;
    line-height: 1;
  }

  a {
    text-decoration: underline;
  }
`;

const NotFound = () => (
  <Background key="404">
    <StyledContainer layout="fullScreen" verticalPadding>
      <Heading fontSize={100}>404</Heading>
      <Link to="/">
        <Heading as="h2">go home</Heading>
      </Link>
    </StyledContainer>
  </Background>
);

export default NotFound;
