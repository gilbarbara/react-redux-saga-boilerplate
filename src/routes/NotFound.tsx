import { Link } from 'react-router-dom';
import { H2, NonIdealState, Page } from '@gilbarbara/components';

import Background from '~/components/Background';

function NotFound() {
  return (
    <Background key="404" data-component-name="NotFound">
      <Page>
        <NonIdealState />
        <Link to="/">
          <H2>go home</H2>
        </Link>
      </Page>
    </Background>
  );
}

export default NotFound;
