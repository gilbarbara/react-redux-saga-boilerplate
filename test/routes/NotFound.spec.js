import React from 'react';
import { mount } from 'enzyme';

import NotFound from 'routes/NotFound';

function setup() {
  return mount(<NotFound />);
}

describe('NotFound', () => {
  const wrapper = setup();

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
