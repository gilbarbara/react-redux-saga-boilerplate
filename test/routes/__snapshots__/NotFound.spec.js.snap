// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render properly 1`] = `"<div class=\\"app__not-found app__route\\"><h1>404</h1></div>"`;
